### ISSUE TITLE
`odo storage create` adds storage to all containers in the devfile

### ISSUE THREAD

**Issue by dharmit** [redhat-developer/odo#4980]
/kind bug
/area storage

## What versions of software are you using?

**Operating System:** Fedora

**Output of `odo version`:** `main`

## How did you run odo exactly?
* created a component from devfile already present in https://github.com/fuse-boosters/fuse-booster-camel-quarkus-ticker2log/tree/next/ (`next` branch)
* added storage using `odo storage create` command

## Actual behavior
odo adds storage to both the containers in the devfile

## Expected behavior
odo should prompt for the container to which storage should be added

## Acceptance Criteria
- [ ] Documentation for `odo storage` command. Along with the usage of the command, it should explain everything mentioned in https://github.com/openshift/odo/issues/4980#issuecomment-895901608
- [ ] `odo describe` command should list the containers in the devfile so that user can use it for `odo url create --container` and `odo storage create --container` commands.

**Comment by kadel** [comment]
> odo adds storage to both the containers in the devfile
This is intentional default behavior.
You can use `--container` flag to specify to which container you want add storage to.

The problem here is that user doesn't know container name unless it checks devfile.yaml 


> odo should prompt for the container to which storage should be added

if you use `odo storage create` with additional flags like `odo storage create  --path /data --size 4Gi` than odo should never go into interactive mode.



`odo storage create` without any additional flags can go into interactive mode, similarly how it is done for `odo create`

`odo storage create mystorage` without `--path` should fail. `--path` needs to be required argument


`odo storage create` commands acts quite strange

`odo storage create` without arguments doesn't fail or go into interactive mode and it creates volume in weird path with random path, not sure how this can be useful for someone
```
    volumeMounts:
    - name: nodejs-nodejs-epwj-xbpc
      path: /nodejs-nodejs-epwj-xbpc

```

specifying just name of the storage `odo storage create mystorage` creates volume in path that is equal to its name, similarly to above, not sure how this can be useful
```
    volumeMounts:
    - name: mystorage
      path: /mystorage
```

**Comment by dharmit** [comment]
> This is intentional default behavior.

Ack. Never knew. Good to know.

> The problem here is that user doesn't know container name unless it checks devfile.yaml

Should there be an odo command that explains devfile in simpler terms? Or maybe `odo describe` could do it?

**Comment by kadel** [comment]
> Or maybe `odo describe` could do it?

that would be a good place for it

**Comment by dharmit** [comment]
/triage needs-information

Adding needs-info for someone to spend time figuring out the scope of issue.

**Comment by kadel** [comment]
no direct storage support in odo v3
/close
